### PR TITLE
Document details mask refinement feature.

### LINKED ISSUE #262: Add docs for the details mask refinement
I would suggest this to be added to the masks->mask refinement & additional controls section above the feathering guide 

************************************************************************************************
details threshold

The details refinement makes the mask more or less valid depending on the amount of detail for every part of the image. The slider allows you to select either areas with lots of detail (slider is positive) or without detail (negative), the default is zero effectively bypassing the refinement.

Please understand that all data for the detail mask refinement are taken from the demosaicer stage in the processing pipeline, not from current image content as when you are using a parametric mask. All your development steps you did on the image will not affect the detail mask itself.
*************************************************************************************************

### PR DESCRIPTION
**Pull request by matt-maguire**
Fixes #262 

A new details mask refinment feature is proposed in PR darktable-org/darktable#8643

This PR is to update the DT docs once that above PR is merged.

### CODE REVIEW

**Comment by elstoc** [comment]
This being quite a new feature, it might be nice to have a few use cases listed as well

File: content/darkroom/masking-and-blending/masks/refinement-controls.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ draft: false
 
 When a parametric or drawn mask is active, several additional sliders are shown which allow the mask to be further refined.
 
+details threshold
+: Details refinement increases or decreases the opacity of the mask depending on the amount of detail in the corresponding part of the image. Use the slider to select either areas with lots of detail (slider is positive) or areas that are flat and lacking in detail (negative). The default is zero, which effectively bypasses the refinement.
+
+**Note:** all the data used for the detail mask refinement is taken from the demosaicer stage in the processing pipeline, not from current image content (which would be the case when defining filtering criteria for a parametric mask). None the development steps coming after the demosaic step will have any effect on calculating the detail mask.
```
Comment (johnny-bit) [inline content/darkroom/masking-and-blending/masks/refinement-controls.md:13]:
```suggestion
**Note:** all the data used for the detail mask refinement is taken from the demosaicer stage in the processing pipeline, not from current image content (which would be the case when defining filtering criteria for a parametric mask). None of the development steps coming after the demosaic step will have any effect on calculating the detail mask.
```

Comment (matt-maguire) [reply content/darkroom/masking-and-blending/masks/refinement-controls.md]: thanks, fixed in the commit below.

**Comment by elstoc** [comment]
Thanks!

**Pull request merged**